### ISSUE TITLE
sdig: Unrecognized HTTPS/SVCB param value can print non-printable characters

### ISSUE THREAD

**Issue by mnordhoff** [PowerDNS/pdns#9829]
- Program: Authoritative
 - Issue type: Bug report

### Short description
If you `sdig` an `HTTPS` record (or probably an `SVCB` record) with an unrecognized parameter, the param value is displayed without being escaped, and can therefore write non-printable characters to your terminal.

It's a (variant of a) character string, so I think it should be `\123` escaped.

### Environment
 - Operating system: Ubuntu 16.04 >_>
 - Software version: 4.5.0+master.0.g2717eea26-1pdns.xenial
 - Software source: PowerDNS repository

### Steps to reproduce
1. `sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse | tail -2 | hd`

### Expected behaviour
Some kind of text output.

### Actual behaviour
```
$ sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse
Reply to question for qname='odoh.cloudflare-dns.com.', qtype=HTTPS
Rcode: 0 (No Error), RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
0       odoh.cloudflare-dns.com.        IN      HTTPS   300     1 . alpn=h2 ipv4hint=104.16.248.249,104.16.249.249 ipv6hint=2606:4700::6810:f8f9,2606:4700::6810:f9f9 key32769=,�(  7_5:�)�*Ƀ�
�f�nD�0��'����P���T
$ sdig ::1 53 odoh.cloudflare-dns.com HTTPS recurse | tail -2 | hd
00000000  30 09 6f 64 6f 68 2e 63  6c 6f 75 64 66 6c 61 72  |0.odoh.cloudflar|
00000010  65 2d 64 6e 73 2e 63 6f  6d 2e 09 49 4e 09 48 54  |e-dns.com..IN.HT|
00000020  54 50 53 09 39 38 09 31  20 2e 20 61 6c 70 6e 3d  |TPS.98.1 . alpn=|
00000030  68 32 20 69 70 76 34 68  69 6e 74 3d 31 30 34 2e  |h2 ipv4hint=104.|
00000040  31 36 2e 32 34 38 2e 32  34 39 2c 31 30 34 2e 31  |16.248.249,104.1|
00000050  36 2e 32 34 39 2e 32 34  39 20 69 70 76 36 68 69  |6.249.249 ipv6hi|
00000060  6e 74 3d 32 36 30 36 3a  34 37 30 30 3a 3a 36 38  |nt=2606:4700::68|
00000070  31 30 3a 66 38 66 39 2c  32 36 30 36 3a 34 37 30  |10:f8f9,2606:470|
00000080  30 3a 3a 36 38 31 30 3a  66 39 66 39 20 6b 65 79  |0::6810:f9f9 key|
00000090  33 32 37 36 39 3d 00 2c  ff 03 00 28 00 20 00 01  |32769=.,...(. ..|
000000a0  00 01 00 20 37 5f 35 3a  e9 29 cc 2a c9 83 d8 0a  |... 7_5:.).*....|
000000b0  01 e2 66 ce 6e 44 c1 30  8b f2 27 94 80 86 d9 50  |..f.nD.0..'....P|
000000c0  f8 d7 e4 54 0a                                    |...T.|
000000c5
```

([`odoh.cloudflare-dns.com` announcement.](https://blog.cloudflare.com/oblivious-dns/))

### Other information
https://tools.ietf.org/html/draft-ietf-dnsop-svcb-https-02#section-2.1
https://tools.ietf.org/html/draft-ietf-dnsop-svcb-https-02#appendix-A

**Comment by mnordhoff** [comment]
I assume this probably also affects things like pdnsutil edit-zone and list-zone. (I didn't remember them until after filing the bug.)

It's possible there's also a bug that it might be unable to correctly *parse* records like this, but I have no idea.

**Comment by pieterlexis** [comment]
side note: `4.5.0+master.0.g2717eea26-1pdns.xenial` seems wrong...

**Comment by pieterlexis** [comment]
it looks like I did not properly implement the unknown SVC params. By the looks of it we need an `XfrText` function that takes a length and does not use 255-byte boundries

**Comment by pieterlexis** [comment]
The fix in #9844 yield this result:

```
$ ./pdns/sdig 8.8.8.8 53 odoh.cloudflare-dns.com HTTPS recurse
Reply to question for qname='odoh.cloudflare-dns.com.', qtype=HTTPS
Rcode: 0 (No Error), RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
0	odoh.cloudflare-dns.com.	IN	HTTPS	299	1 . alpn=h2 ipv4hint=104.16.248.249,104.16.249.249 ipv6hint=2606:4700::6810:f8f9,2606:4700::6810:f9f9 key32769="\000,\255\003\000(\000 \000\001\000\001\000 #\252\014TK[\180\021v\022h\022\140c\251]\248X\198\207\236\152]\173?1\230T\129\244\232m"
```
